src/Sulu/Component/CustomUrl/Routing/CustomUrlDefaultsProvider.php

```diff
@@ -0,0 +1,116 @@
+{
+    public function __construct(
+        private DocumentManagerInterface $documentManager,
+        private DocumentInspector $inspector,
+        private StructureManagerInterface $structureManager,
+        private WebspaceManagerInterface $webspaceManager
+    ) {
+    }
+
+    public function provideDefault(Request $request, CustomUrl $customUrl): array
+    {
+        $seo = [
+            'noFollow' => $customUrl->isNoFollow(),
+            'noIndex' => $customUrl->isNoIndex(),
+        ];
+
+        if ($customUrl->isCanonical()) {
+            $document = $this->documentManager->find($customUrl->getTargetDocument());
+            $resourceSegment = $document->getResourceSegment();
+            $seo['canonicalUrl'] = $this->webspaceManager->findUrlByResourceLocator(
+                $resourceSegment,
+                null,
+                $customUrl->getTargetLocale(),
+                $customUrl->getWebspace(),
+                $request->getHost(),
+                $request->getScheme()
+            );
+        }
+
+        return [
+            '_seo' => $seo,
+            '_webspace' => $this->webspaceManager->findWebspaceByKey($customUrl->getWebspace()),
+            '_environment' => 'dev',
+        ];
+    }
+
+    public function provideForRedirect(Request $request, CustomUrl $customUrl): array
+    {
+        $resourceSegment = '/';
+        if (null !== $customUrl->getTargetDocument()) {
+            $document = $this->documentManager->find($customUrl->getTargetDocument());
+            $resourceSegment = $document->getResourceSegment();
+        }
+
+        $url = $this->webspaceManager->findUrlByResourceLocator(
+            resourceLocator: $resourceSegment,
+            environment: null,
+            languageCode: $customUrl->getTargetLocale(),
+            webspaceKey: null,
+            domain: $request->getHost(),
+            scheme: $request->getScheme()
+        );
+
+        $requestFormat = $request->getRequestFormat(null);
+        $requestFormatSuffix = $requestFormat ? '.' . $requestFormat : '';
+
+        $queryString = $request->getQueryString();
+        $queryStringSuffix = $queryString ? '?' . $queryString : '';
+
+        return [
+            '_controller' => SuluRedirectController::class . '::redirectAction',
+            'url' => $url . $requestFormatSuffix . $queryStringSuffix,
+        ];
+    }
+
+    public function provideForForward(Request $request, CustomUrl $customUrl): array
+    {
+        $document = $this->documentManager->find($customUrl->getTargetDocument());
+
+        //if ($document->getNodeType() === Structure::NODE_TYPE_EXTERNAL_LINK) {
+        //return [
+        //'_controller' => [SuluRedirectController, 'redirectAction'],
+        //'url' => $document->getResourceLocator(),
+        //];
+        //} else if ($document->getNodeType() === Structure::NODE_TYPE_INTERNAL_LINK) {
+        //$structure = $document->getInternalLinkContent();
+        //} else {
+        //}
+
```

Comment:
This should be the same logic as it was before with the `ExternalLink` and `InternalLink` from the Enhancers directory.